cparser/Elab.ml

```diff
@@ -983,13 +1002,18 @@ let init_int_array_wstring opt_size s =
 let check_init_type loc env a ty =
   if wrap2 valid_assignment loc env a ty then ()
   else if wrap2 valid_cast loc env a.etyp ty then
-    warning loc
-      "initializer has type@ %a@ instead of the expected type @ %a"
-      Cprint.typ a.etyp Cprint.typ ty
+    if wrap2 int_pointer_conversion loc env a.etyp ty then
+      warning loc Int_conversion
+        "incompatible integer-pointer conversion initializer has type %a instead of the expected type %a"
```

Comment:
I'd put a colon for readability: "incompatible integer-pointer conversion**:** initializer "